src/addons/plugin/addon-manager-proxy.js

```diff
@@ -80,15 +83,25 @@ class AddonManagerProxy extends EventEmitter {
    */
   onMsg(msg) {
     DEBUG && console.log('AddonManagerProxy: Rcvd:', msg);
-    var adapter = this.adapter;
+
+    // The first switch covers unload plugin.
+    switch (msg.messageType) {
+      case Constants.UNLOAD_PLUGIN:
+        this.unloadPlugin();
+        return;
+    }
+
+    // The second switch covers adapter messages. i.e. don't have a deviceId.
+    // or don't need a device object.
+
+    var adapterId = msg.data.adapterId;
+    var adapter = this.adapters.get(adapterId);
     if (!adapter) {
-      console.error('AddonManagerProxy: No adapter added yet.')
+      console.error('AddonManagerProxy: Unregcognized adapter:', adapterId);
```

Comment:
Typo: `unrecognized`

Reply:
Thank you for the review!
I fixed Typo.
Since plugin.js also had the same Typo, I fixed it.
If it is unnecessary, I rebase the comitt.